luxonis_train/callbacks/README.md

```diff
@@ -65,3 +65,16 @@ Callback to perform a test run at the end of the training.
 ## `UploadCheckpoint`
 
 Callback that uploads currently the best checkpoint (based on validation loss) to the tracker location - where all other logs are stored.
+
+## `GradCamCallback`
+
+Callback to visualize gradients using Grad-CAM. Works only during validation.
+
+**Parameters:**
+
+| Key             | Type  | Default value      | Description                                          |
+| --------------- | ----- | ------------------ | ---------------------------------------------------- |
+| `target_layer`  | `int` | -                  | Layer to visualize gradients.                        |
+| `class_idx`     | `int` | 0                  | Index of the class for visualization. Defaults to 0. |
```

Comment:
Should we make it we visualize for all classes by default (e.g. if -1) and then for specific one if set to some positive integer? Also we'll want to check if this number is > than number of all classes.

Reply:
Initially, I had this option, but I decided to remove it because typically we want to examine/debug a specific class. If all classes are visualized, we can gain some insights, but it would be harder to understand where the issue lies. Moreover, adding more target classes will increase the callback processing time, as it will require more forward passes instead of a single batched pass due to the implementation of the Grad-CAM library. We can also add visualization of gradients separately for each or all target classes. However, I think we can keep advanced functionality for the future if it becomes necessary. And that’s a good point about the check.